mobile/src/calendar/days-counters/days-counters.tsx

```diff
@@ -11,10 +11,17 @@ interface DaysCountersProps {
     daysCounters: DaysCountersModel;
 }
 
-class DaysCountersImpl extends Component<DaysCountersProps> {
+interface ExplicitDaysCountersProps {
+    explicitCounters?: DaysCountersModel;
+    explicitStyle?: StyleProp<ViewStyle>;
+}
+
+class DaysCountersImpl extends Component<DaysCountersProps & ExplicitDaysCountersProps> {
 
     public render() {
-        const { daysCounters: { allVacationDays, hoursCredit } } = this.props;
+        const { allVacationDays, hoursCredit } = this.props.explicitCounters ?
+            this.props.explicitCounters :
+            this.props.daysCounters;
 
         if (!allVacationDays && !hoursCredit) {
```

Comment:
Are you sure this condition will never match in case of explicit props passed? Cause it will be infinite in this case.